sklearn/linear_model/_ridge.py

```diff
@@ -777,9 +788,9 @@ def _ridge_regression(
                 coef[i] = coef_
             n_iter[i] = n_iter_
 
+        intercept = xp.asarray(intercept)
```

Comment:
I think it would be good to add a short comment that explains why we are doing something with `xp` here. Naively you'd think that because Array API only supports SVD as solver, you'd never come into this branch of code. And there are a few lines above that use `np.`. I think it can happen that we use "Array API dispatching" and a Numpy array and SAG solver and then come into this code, but that is kinda of a guess.

So I think it would be good to leave a note to our future selves that explains why this code isn't a typo :D